src/app/components/single-hero-page/single-hero-page.component.ts

```diff
@@ -0,0 +1,27 @@
+import { HeroModel } from './../../models/heroModel';
+import { HeroService } from '../../services/single-page/hero-service';
+import { Component, OnInit } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+
+@Component({
+  selector: 'app-single-hero-page',
+  templateUrl: './single-hero-page.component.html',
+  styleUrls: ['./single-hero-page.component.scss']
+})
+export class SingleHeroPageComponent implements OnInit {
+  private _hero: HeroModel;
+  private _id: number;
+
+  constructor(private _getSingleHero: HeroService, private _route: ActivatedRoute ) {
+    this._id = Number(this._route.snapshot.paramMap.get('id'));
+    this._getSingleHero.getHero(this._id)
+      .subscribe(data => this._hero = data);
+  }
+
+  ngOnInit() {
+  }
+
+  private get hero(): HeroModel {
```

Comment:
this seems unnecessary, its private and it only returns the hero, you might as well use the variable instead of this function

Reply:
My intention was to have the hero as private, and if i don't have the getter than the html doesn't see it. For now i deleted the getter, and put the hero on public. Does the hero have to be private?